DataFormats/CaloTowers/src/CaloTower.cc

```diff
@@ -25,7 +28,11 @@ CaloTower::CaloTower(const CaloTowerDetId& id,
       hadE_(hadE),
       outerE_(outerE),
       emLvl1_(ecal_tp),
-      hadLvl1_(hcal_tp) {}
+      hadLvl1_(hcal_tp) {
+  subdet_ = HcalEmpty;
+  inHO_ = false;
+  inHBHEgap_ = false;
```

Comment:
```suggestion
      hadLvl1_(hcal_tp),
      subdet_(HcalEmpty),
      inHO_(false),
      inHBHEgap_(false) {
```
the initializer list is usually preferred over the body of the constructor.

Reply:
my suggestion is as displayed; a more proper phrasing is to use initialization in the member initializer list instead of using assignment in the constructor body
e.g. https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c49-prefer-initialization-to-assignment-in-constructors

Reply:
it's perhaps even better to initialize in the member declaration part in CaloTower.h

Reply:
OK,  mea culpa (misunderstanding) ...